### DISCUSSION TITLE
How to set a fixed value for the sw8-correlation field of the agent request header

### DISCUSSION THREAD

**Discussion by bongmu** [apache/skywalking#7562] [Q&A] [answered]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
___
### Question
- What do you want to know?
HI,
In order to adapt to our system, I need to modify on the basis of the skywalking agent source code, and set the sw8-correlation field to a fixed value. How do I modify it and which file should I modify? Can anyone guide me?

**Comment by wu-sheng** [comment, marked as answer]
I am not sure what do you mean adopting your system, could you be more clear?

**Reply by bongmu** [reply]
When the java agent connects to the skywalking server incorrectly, the key of the header header will carry namespace+sw8, namespace-sw8-correlation, two fields, but the value is empty. This causes our java microservices to access some APIs, some APIs will be strictly verified, and an error will be reported if the value is empty.

Should I need to modify the source code of the agent, can you guide me how to modify which file?

**Reply by wu-sheng** [reply]
I think you can't easily change this. This is a core level feature.
Which project, document and protocol have the rules that header value must be not empty?

**Reply by bongmu** [reply]
It’s the api interface of one of our customers, we can’t change it

**Reply by wu-sheng** [reply]
The only temp fix in my mind is removing the whole `correlation` header.

**Reply by bongmu** [reply]
thanks，Yeah is very happy to hear that there is a temporary repair method, can you tell me which file code should be modified?

**Reply by wu-sheng** [reply]
Remove `SW8CorrelationCarrierItem` from `ContextCarrier#items()`.

**Reply by bongmu** [reply]
located  apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java 

 remove The following code in public CarrierItem items() ：

SW8CorrelationCarrierItem sw8CorrelationCarrierItem = new SW8CorrelationCarrierItem(
            correlationContext, sw8ExtensionCarrierItem);

Is it?

**Comment by bongmu** [comment]
The problem has been solved, thank you
